fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/dynamic/DynamicSchema.java

```diff
@@ -0,0 +1,427 @@
+ * generated.
+ */
+public class DynamicSchema {
+    public static final DynamicSchema EMPTY_SCHEMA = empty();
+
+    private final FileDescriptorSet fileDescSet;
+    private final Map<String,Descriptor> msgDescriptorMapFull = new HashMap<>();
+    private final Map<String,Descriptor> msgDescriptorMapShort = new HashMap<>();
+    private final Map<String,EnumDescriptor> enumDescriptorMapFull = new HashMap<>();
+    private final Map<String,EnumDescriptor> enumDescriptorMapShort = new HashMap<>();
+    private final Map<Type, String> typeToNameMap;
+
+    // --- public static ---
+    private static String duplicateNameErrorMessage = "duplicate name: %s";
+
+    public static DynamicSchema empty() {
+        FileDescriptorSet.Builder resultBuilder = FileDescriptorSet.newBuilder();
+        try {
+            return new DynamicSchema(resultBuilder.build(), Maps.newHashMap());
+        } catch (final DescriptorValidationException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    /**
+     * Creates a new dynamic schema builder.
+     *
+     * @return the schema builder
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    @SuppressWarnings("PMD.AssignmentInOperand")
+    public static DynamicSchema parseFrom(InputStream schemaDescIn) throws DescriptorValidationException, IOException {
+        try (schemaDescIn) {
+            int len;
+            byte[] buf = new byte[4096];
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            while ((len = schemaDescIn.read(buf)) > 0) {
+                baos.write(buf, 0, len);
+            }
+            return parseFrom(baos.toByteArray());
+        }
+    }
+
+    public static DynamicSchema parseFrom(byte[] schemaDescBuf) throws DescriptorValidationException, IOException {
+        return new DynamicSchema(FileDescriptorSet.parseFrom(schemaDescBuf), Maps.newHashMap());
+    }
+
+    // --- public ---
+
+    /**
+     * Creates a new dynamic message builder for the given message type.
+     *
+     * @param msgTypeName the message type name
+     * @return the message builder (null if not found)
+     */
+    public DynamicMessage.Builder newMessageBuilder(String msgTypeName) {
+        Descriptor msgType = getMessageDescriptor(msgTypeName);
+        if (msgType == null) {
+            return null;
+        }
+        return DynamicMessage.newBuilder(msgType);
+    }
+
+    /**
+     * Creates a new dynamic message builder for the given type.
+     *
+     * @param type the type name
+     * @return the message builder (null if not found)
+     */
+    public DynamicMessage.Builder newMessageBuilder(@Nonnull final Type type) {
+        final String msgTypeName = typeToNameMap.get(type);
+        Objects.requireNonNull(msgTypeName);
+        return newMessageBuilder(msgTypeName);
+    }
+
+    public String getProtoTypeName(@Nonnull final Type type) {
```

Comment:
This whole class needs nullability annotations.

Reply:
done.